app/src/main/java/cz/quanti/razym/rocketapp/system/RocketDetailScreen.kt

```diff
@@ -0,0 +1,541 @@
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.navArgument
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import cz.quanti.razym.rocketapp.R
+import cz.quanti.razym.rocketapp.model.RocketDetail
+import cz.quanti.razym.rocketapp.model.Stage
+import cz.quanti.razym.rocketapp.presentation.RocketDetailViewModel
+import cz.quanti.razym.rocketapp.ui.theme.LocalColors
+import cz.quanti.razym.rocketapp.ui.theme.RocketappTheme
+import org.koin.androidx.compose.getViewModel
+
+
+private const val ROCKET_DETAIL = "rocketDetail"
+
+private const val ROCKET_ID = "rocketId"
+
+data object RocketDetailScreen : Screen(
+    route = "$ROCKET_DETAIL/{$ROCKET_ID}",
+    navArguments = listOf(
+        navArgument(ROCKET_ID) { type = NavType.StringType },
+    )
+) {
+    fun createRoute(rocketId: String) = "$ROCKET_DETAIL/${rocketId}"
+}
+
+fun NavGraphBuilder.rocketDetailScreen(
+    onBackClick: () -> Unit = {},
+    onLaunchClick: () -> Unit = {},
+) {
+    composable(
+        route = RocketDetailScreen.route,
+        arguments = RocketDetailScreen.navArguments,
+    ) {
+        val viewModel: RocketDetailViewModel = getViewModel()
+        val uiState by viewModel.uiState.collectAsState()
+
+        val rocketId = it.arguments!!.getString(RocketDetailScreen.navArguments[0].name)!!
+
+        LaunchedEffect(rocketId) {
+            // TODO this is not perfect condition, as after configuration change
+            //  it will reload data if there has been an error and there are no rockets then.
+            if (!uiState.loading && uiState.rocket == null) {
+                viewModel.fetchRocket(rocketId)
+            }
+        }
+
+        RocketDetailScreen(
+            uiState = uiState,
+            onBackClick = onBackClick,
+            onLaunchClick = onLaunchClick,
+            onErrorClick = { viewModel.fetchRocket(rocketId) },
+        )
+    }
+}
+
+fun NavController.navigateToRocketDetail(rocketId: String, navOptions: NavOptions? = null) {
+    this.navigate(
+        route = RocketDetailScreen.createRoute(rocketId),
+        navOptions = navOptions
+    )
+}
+
+@Composable
+fun RocketDetailScreen(
+    uiState: RocketDetailViewModel.ScreenUiState,
+    onBackClick: () -> Unit = {},
+    onLaunchClick: () -> Unit = {},
+    onErrorClick: () -> Unit = {},
+) {
+    val rocket = uiState.rocket
+    val loading = uiState.loading
+
+    RocketDetailScreen(loading, rocket, onBackClick, onLaunchClick, onErrorClick)
+}
+
+@Composable
+private fun RocketDetailScreen(
+    loading: Boolean,
+    rocket: RocketDetail?,
+    onBackClick: () -> Unit = { },
+    onLaunchClick: () -> Unit = { },
+    onErrorClick: () -> Unit = { },
+) {
+    Scaffold(
+        topBar = {
+            TopBar(
+                // TODO, because we does not persist data, use a placeholder for title
+                //  when not loaded yet.
+                title = rocket?.name ?: "Loading",
```

Comment:
use string resource for Loading or send the title as a navigation parameter

Reply:
used both